pkg/internal/testhelpers/postgres_container.go

```diff
@@ -49,95 +47,80 @@ var (
+type TestOptions struct {
+	options              OptionState
+	timescaleDockerImage string
+}
+
+func NewTestOptions(options OptionState, timescaleDockerImage string) TestOptions {
+	return TestOptions{
+		options,
+		timescaleDockerImage,
+	}
+}
 
 const (
 	timescaleBit        = 1 << iota
 	multinodeBit        = 1 << iota
-	postgres12Bit       = 1 << iota
-	postgres13Bit       = 1 << iota
 	timescaleNightlyBit = 1 << iota
 )
 
 const (
-	Timescale ExtensionState = timescaleBit
-	Multinode ExtensionState = timescaleBit | multinodeBit
+	Timescale OptionState = timescaleBit
+	Multinode OptionState = timescaleBit | multinodeBit
 
-	TimescaleNightly          ExtensionState = timescaleBit | timescaleNightlyBit
-	TimescaleNightlyMultinode ExtensionState = timescaleBit | multinodeBit | timescaleNightlyBit
+	TimescaleNightly          OptionState = timescaleBit | timescaleNightlyBit
+	TimescaleNightlyMultinode OptionState = timescaleBit | multinodeBit | timescaleNightlyBit
 )
 
-func (e *ExtensionState) UseTimescaleDB() {
-	*e |= timescaleBit
-}
-
-func (e *ExtensionState) UseTimescaleNightly() {
-	*e |= timescaleBit | timescaleNightlyBit
+func (e *TestOptions) UseTimescaleDB() {
+	e.options |= timescaleBit
 }
 
-func (e *ExtensionState) UseTimescaleNightlyMultinode() {
-	*e |= timescaleBit | multinodeBit | timescaleNightlyBit
+func (e *TestOptions) UseTimescaleNightly() {
+	e.options |= timescaleBit | timescaleNightlyBit
 }
 
-func (e *ExtensionState) UseMultinode() {
-	*e |= timescaleBit | multinodeBit
+func (e *TestOptions) UseTimescaleNightlyMultinode() {
+	e.options |= timescaleBit | multinodeBit | timescaleNightlyBit
 }
 
-func (e *ExtensionState) UsePG12() {
-	*e |= postgres12Bit
+func (e *TestOptions) UseMultinode() {
+	e.options |= timescaleBit | multinodeBit
 }
 
-func (e *ExtensionState) UsePG13() {
-	*e |= postgres13Bit
+func (e *TestOptions) SetTimescaleDockerImage(timescaleDockerImage string) {
+	e.timescaleDockerImage = timescaleDockerImage
 }
 
-func (e ExtensionState) UsesTimescaleDB() bool {
-	return (e & timescaleBit) != 0
+func (e TestOptions) UsesTimescaleDB() bool {
+	return (e.options & timescaleBit) != 0
 }
 
-func (e ExtensionState) UsesTimescaleNightly() bool {
-	return (e & timescaleNightlyBit) != 0
+func (e TestOptions) UsesTimescaleNightly() bool {
+	return (e.options & timescaleNightlyBit) != 0
 }
 
-func (e ExtensionState) UsesMultinode() bool {
-	return (e & multinodeBit) != 0
+func (e TestOptions) UsesMultinode() bool {
+	return (e.options & multinodeBit) != 0
 }
 
-func (e ExtensionState) UsesPG12() bool {
-	return (e & postgres12Bit) != 0
+func (e TestOptions) GetDockerImageName() string {
+	return e.timescaleDockerImage
 }
 
-func (e ExtensionState) UsesPG13() bool {
-	return (e & postgres13Bit) != 0
-}
-
-func (e ExtensionState) GetPGMajor() string {
-	PGMajor := "14"
-	if e.UsesPG13() {
-		PGMajor = "13"
-	} else if e.UsesPG12() {
-		PGMajor = "12"
+func (e TestOptions) TryGetDockerImagePgVersion() (string, error) {
```

Comment:
can we exec into the image and get the version from the pg_config binary instead?

Reply:
Yes, but only after the container has been started. In the upgrade tests (where this is used), there is no `timescaledev/promscale-extension:0.1.2-ts2-pg14` docker image, so trying to start it will fail. I think that we need to rework the upgrade tests a little anyway to make it a bit smoother.